warehouse/oidc/services.py

```diff
@@ -23,6 +24,24 @@
 from warehouse.oidc.interfaces import IOIDCProviderService
 
 
+class InsecureOIDCProviderWarning(UserWarning):
+    pass
+
+
+@implementer(IOIDCProviderService)
+class NullOIDCProviderService:
+    def __init__(self, provider, issuer_url, cache_url, metrics):
+        warnings.warn(
+            "NullOIDCProviderService is intended only for use in development, "
+            "you should not use it in production due to the lack of actual "
+            "JWT verification.",
+            InsecureOIDCProviderWarning,
+        )
+
+    def verify_for_project(self, token, project):
+        return True
```

Comment:
NB: I'm going to change this so that it actually checks `project.oidc_providers`. It still won't do the actual JWT signature verification, however.